src/browser/BrowserOptionDialog.cpp

```diff
@@ -102,3 +112,16 @@ void BrowserOptionDialog::saveSettings()
     settings.setFirefoxSupport(m_ui->firefoxSupport->isChecked());
     settings.setVivaldiSupport(m_ui->vivaldiSupport->isChecked());
 }
+
+void BrowserOptionDialog::showProxyLocationFileDialog()
+{
+#ifdef Q_OS_WIN
+    QString fileTypeFilter(tr("Executable Files (*.exe);;All Files (*.*)"));
+#else
+    QString fileTypeFilter(tr("Executable Files (*.*)"));
+#endif
+    auto proxyLocation = QFileDialog::getOpenFileName(this, tr("Select custom proxy location"),
```

Comment:
Please use `FileDialog::getOpenFileName()`

Reply:
I must admit I forgot about that class. But I don't think I should use it here. The main purpose of FileDialog is to remember the last open/save location. We don't want that here. In fact, it would be very annoying if the next time I click "Open Database" the selected directory would be the KeePassXC install folder.

Reply:
You can use `FileDialog::setNextForgetDialog()` before calling `getOpenFileName()` to not save the final destination, and you can specify the `dir` argument so you won't be placed in the `LastDir`.

Using our FileDialog implementation helps with future tests (since implements `nextFileName`), fixes focus on macOS and if in the future we will modify something that will get propagated here as well

Reply:
I set a few additional flags for the QFileDialog, which can't be set via the static method, so I had to change this to a custom instantiation of QFileDialog. That makes it impossible to use FileDialog here. I therefore added the extra activateWindow() fix and would suggest to postpone this to a refactor of FileDialog, during which we make it a subclass of QFileDialog.